public/fetch.js

```diff
@@ -1,6 +1,8 @@
-const xhr = new XMLHttpRequest()
 
 const fetch = (url, cb)=>{
+    const xhr = new XMLHttpRequest();
+    let searchParam = searchInput.value; 
```

Comment:
Here, you should seperate the fetching logic from any dom implementation, searchParam value should be generic(passed as an argument to the fetch function along with the url)
```js
fetch(`${url}?q=${searchParam}`, console.log)
```

As you can see, it should be sent along in the querystring not as a custom header